### PR TITLE
Add a BBE for isolated functions

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

### CODE REVIEW

File: examples/isolated-functions/isolated_functions.bal [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import ballerina/io;
+
+const BALLERINA = "Ballerina";
+
+final string[] & readonly prefixes = ["hello", "greetings", "hello world"];
+
+// An `isolated` function that adds to an array a specific number of
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.bal:7]:
```suggestion
// An `isolated` function, which adds a specific number of
```

File: examples/isolated-functions/isolated_functions.bal [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// greeting strings.
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.bal:8]:
```suggestion
// greeting strings to an array.
```

File: examples/isolated-functions/isolated_functions.bal [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    // Call another `isolated` function to build a specific number
+    // of greetings.
+    string[] builtGreetings = getGreetings(count);
+
+    // Add the greetings returned by the `isolated` function to
+    // the mutable array passed as an argument.
+    greetings.push(...builtGreetings);
+}
+
+// An `isolated` function that accesses global immutable state and
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.bal:22]:
```suggestion
// An `isolated` function, which accesses the global, immutable state and
```

File: examples/isolated-functions/isolated_functions.bal [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        builtGreetings.push(prefix + " from " + BALLERINA);
+    }
+
+    return builtGreetings;
+}
+
+string[] mutableGreetingArray = [];
+
+public function main() {
+    // Access global mutable state `mutableGreetingArray`, and
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.bal:39]:
```suggestion
    // Access the global, mutable `mutableGreetingArray` state and
```

File: examples/isolated-functions/isolated_functions.bal [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    // pass it as an argument to an `isolated` function.
+    // Mutable state can be passed as an argument to an `isolated`
+    // function. But this function which accesses global mutable state
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.bal:42]:
```suggestion
    // function. However, this function, which accesses the global,mutable state
```

File: examples/isolated-functions/isolated_functions.description [hunk 6]
```diff
@@ -0,0 +1,4 @@
+// An `isolated` function is a function that accesses mutable state only via its parameters.
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.description:1]:
```suggestion
// An `isolated` function is a function that accesses a mutable state only via its parameters.
```

File: examples/isolated-functions/isolated_functions.description [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+// Such a function can only access module level state if it is final and immutable. Any
```
Comment (praneesha) [inline examples/isolated-functions/isolated_functions.description:2]:
```suggestion
// Such a function can only access a module-level state if it is final and immutable. Any
```

File: examples/isolated-functions/isolated_functions.bal (hunk 3 above)
Comment (MaryamZi) [reply examples/isolated-functions/isolated_functions.bal]: State in this context is not singular. Applicable to the other suggestions too.

**Comment by niveathika** [comment]
@MaryamZi Please add meta tags file

File: examples/isolated-functions/isolated_functions.bal (hunk 3 above)
Comment (praneesha) [reply examples/isolated-functions/isolated_functions.bal]: +1

File: examples/isolated-functions/isolated_functions.description (hunk 6 above)
Comment (MaryamZi) [reply examples/isolated-functions/isolated_functions.description:1]: Resolving these as per the offline discussion.

**Review by praneesha — APPROVED** [review]

File: examples/isolated-functions/isolated_functions.bal [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    // of greetings.
+    string[] builtGreetings = getGreetings(count);
+
+    // Add the greetings returned by the `isolated` function to
+    // the mutable array passed as an argument.
+    greetings.push(...builtGreetings);
+}
+
+// An `isolated` function, which accesses global immutable state and
+// returns a mutable value.
```
Comment (pubudu91) [inline examples/isolated-functions/isolated_functions.bal:23]: You mentioned that if locks are used, we can access global mutable state right? If so, shouldn't that be mentioned?

Comment (MaryamZi) [reply examples/isolated-functions/isolated_functions.bal:23]: Yeah, but that is for [isolated module level variables](https://github.com/ballerina-platform/ballerina-lang/issues/25898) which we don't support yet.

**Pull request merged**